rdfunit-core/src/main/java/org/aksw/rdfunit/sources/SchemaSource.java

```diff
@@ -1,51 +1,66 @@
 package org.aksw.rdfunit.sources;
 
+import com.google.common.collect.Lists;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
 import lombok.ToString;
 import org.aksw.rdfunit.enums.TestAppliesTo;
+import org.aksw.rdfunit.io.reader.RdfMultipleReader;
 import org.aksw.rdfunit.io.reader.RdfReader;
 import org.aksw.rdfunit.io.reader.RdfReaderException;
 import org.apache.jena.ontology.OntModel;
 import org.apache.jena.ontology.OntModelSpec;
-import org.apache.jena.rdf.model.Model;
-import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.*;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.*;
+import java.util.stream.Collectors;
+
 /**
  * @author Dimitris Kontokostas
  * @since 9/16/13 1:51 PM
  */
 @ToString
-@EqualsAndHashCode(exclude={"model", "schemaReader"})
+@EqualsAndHashCode(exclude={"model", "schemaReader", "imports"})
```

Comment:
Yes, sounds good. I will create the subclass.

Reply:
one approach would be to get all the schema sources that are defined explicitly (or the ones we got automatically).
We iterate over all of them and we get all transitive `owl:imports` from each schema source instance. 
We gather all import IRIs into a set (for deduplication) and then reconcile them with the schema service.

at the end, we return the original schema sources along with the ones we computed and a new collection for input to RDFnit

Reply:
This is basically already implemented, except that I load each unique import into one model.
So you would like to change the interface like this? 

public Model getTransitiveModel() -> public Collection<SchemaSource> getTransitiveSchemaSources()

Where would use it then? Atm I only use it in TagRdfUnitTestGenerator...

Reply:
I was thinking to use this a bit differently so that people use the owl:import computation on demand.

instead of creating a separate class that is fixed in the code, I was thinking of a (static) function that takes the original 
SchemaSource list and returns an augmented list back and this function would be called in the `RDFUnitConfiguration.getAllSchemata()`.

This way we could even have a parameter specifying if we want to compute the owl:imports or not in the future, or maybe allow to get sources even not defined in the SchemaService.

So, that function would look something like the following (not tested):

```
    public static List<SchemaSource> augmentWithOwlImports(List<SchemaSource> originalSources) {

        ImmutableList.Builder<SchemaSource> augmentedSources = ImmutableList.builder();
        augmentedSources.addAll(originalSources);

        Set<String> schemaIris = originalSources.stream().map(SchemaSource::getSchema).collect(Collectors.toSet());
        Set<SchemaSource> currentSources = new HashSet<>(originalSources);

        while (!currentSources.isEmpty()) {

            Set<SchemaSource> computedSources = currentSources.stream()
                    .map(SchemaSource::getModel)
                    .flatMap(m -> m.listObjectsOfProperty(OWL.imports).toList().stream())
                    .filter(RDFNode::isResource)
                    .map(RDFNode::asResource)
                    .map(Resource::getURI)
                    .filter(uri -> !schemaIris.contains(uri))
                    .map(SchemaService::getSourceFromUri)
                    .filter(Optional::isPresent)
                    .map(Optional::get)
                    .collect(Collectors.toSet());

            Set<String> newIris = computedSources.stream().map(SchemaSource::getSchema).collect(Collectors.toSet());
            schemaIris.addAll(newIris);

            augmentedSources.addAll(computedSources);
            
            currentSources = computedSources;

        }

        return augmentedSources.build();
    }
```

wdyt?

Reply:
Implemented as stated (worked without changing a line 👍 ), introduced an -i option in the cli